src/Orleans.Serialization/Codecs/IntegerCodec.cs

```diff
@@ -745,4 +749,210 @@ public static ulong ReadValue<TInput>(ref Reader<TInput> reader, Field field)
+            }
+            else
+            {
+                ReferenceCodec.MarkValueField(writer.Session);
+                SignedBinaryIntegerCodec<Int128>.WriteField(ref writer, fieldIdDelta, expectedType, in value);
+            }
+        }
+
+        /// <inheritdoc/>
+        Int128 IFieldCodec<Int128>.ReadValue<TInput>(ref Reader<TInput> reader, Field field) => ReadValue(ref reader, field);
+
+        /// <summary>
+        /// Reads a value.
+        /// </summary>
+        /// <typeparam name="TInput">The reader input type.</typeparam>
+        /// <param name="reader">The reader.</param>
+        /// <param name="field">The field.</param>
+        /// <returns>The value.</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static Int128 ReadValue<TInput>(ref Reader<TInput> reader, Field field)
+        {
+            if (field.WireType != WireType.LengthPrefixed)
+            {
+                return Int64Codec.ReadValue(ref reader, field);
+            }
+
+            ReferenceCodec.MarkValueField(reader.Session);
+            Int128 result;
+            SignedBinaryIntegerCodec<Int128>.ReadLengthPrefixed(ref reader, out result);
+            return result;
+        }
+
+        /// <summary>
+        /// Reads a value without protocol framing.
+        /// </summary>
+        /// <typeparam name="TInput">The reader input type.</typeparam>
+        /// <param name="reader">The reader.</param>
+        /// <returns>The value.</returns>
+        internal static Int128 ReadRaw<TInput>(ref Reader<TInput> reader)
+        {
+            Int128 result;
+            SignedBinaryIntegerCodec<Int128>.ReadLengthPrefixed(ref reader, out result);
+            return result;
+        }
+    }
+
+    /// <summary>
+    /// Serializer for <see cref="UInt128"/>.
+    /// </summary>
+    [RegisterSerializer]
+    public sealed class UInt128Codec : IFieldCodec<UInt128>
+    {
+        private static readonly Type CodecFieldType = typeof(UInt128);
+
+        /// <inheritdoc/>
+        void IFieldCodec<UInt128>.WriteField<TBufferWriter>(ref Writer<TBufferWriter> writer, uint fieldIdDelta, Type expectedType, UInt128 value) => WriteField(ref writer, fieldIdDelta, expectedType, value);
+
+        /// <summary>
+        /// Writes a field.
+        /// </summary>
+        /// <typeparam name="TBufferWriter">The buffer writer type.</typeparam>
+        /// <param name="writer">The writer.</param>
+        /// <param name="fieldIdDelta">The field identifier delta.</param>
+        /// <param name="expectedType">The expected type.</param>
+        /// <param name="value">The value.</param>
+        public static void WriteField<TBufferWriter>(ref Writer<TBufferWriter> writer, uint fieldIdDelta, Type expectedType, UInt128 value) where TBufferWriter : IBufferWriter<byte>
+        {
+            if (value <= (UInt128)ulong.MaxValue && value >= (UInt128)ulong.MinValue)
```

Comment:
No need for `value >= (UInt128)ulong.MinValue` (it's always true).

Reply:
Thanks for catching that. Silly mistake